module/VuFind/tests/unit-tests/src/VuFindTest/OAuth2/Repository/AbstractTokenRepositoryTestCase.php

```diff
@@ -355,24 +299,42 @@ protected function getMockAccessTokenService(): AccessTokenServiceInterface
         return $accessTokenService;
     }
 
+    /**
+     * Create User entity mock.
+     *
+     * @param int    $id       User ID
+     * @param string $username User's name
+     *
+     * @return MockObject&UserEntityInterface
+     */
+    protected function createMockUserEntity(int $id, string $username): UserEntityInterface&MockObject
+    {
+        $mockUser = $this->createMock(UserEntityInterface::class);
+        $mockUser->expects($this->any())
+            ->method('getId')
+            ->willReturn($id);
+        $mockUser->expects($this->any())
+            ->method('getUsername')
+            ->willReturn($username);
+        return $mockUser;
+    }
+
     /**
      * Create User service
      *
      * @return MockObject&UserServiceInterface
      */
-    protected function getMockUserService(): UserServiceInterface
+    protected function getMockUserService(): UserServiceInterface&MockObject
     {
-        $userTable = $this->getMockUserTable();
-        $userService = $this->createMock(UserServiceInterface::class);
-        $userService->expects($this->any())
+        $mockUserService = $this->createMock(UserServiceInterface::class);
+        $mockUserService->expects($this->any())
             ->method('getUserByField')
-            ->willReturnCallback(
-                function ($fieldName, $fieldValue) use ($userTable) {
-                    $this->assertEquals('id', $fieldName);
-                    return $userTable->getById($fieldValue);
-                }
-            );
-        return $userService;
+            ->willReturnCallback(function (string $fieldName, $fieldValue) {
+                $this->assertEquals('id', $fieldName);
+                return $this->createMockUserEntity(2, 'test');
```

Comment:
I think some of the failing tests may be caused by the hard-coded ID of 2 here -- should this be `$fieldValue` instead?